### DISCUSSION TITLE
How to create a shareable link?

### DISCUSSION THREAD

**Discussion by jsalgado10** [OrchardCMS/OrchardCore#7640] [Q&A]
I am trying to add a way where the user can create a shareable link to a specific document with an expiration date. Currently only authenticated users can view content. I want the user to be able to share a document with the public and maybe set an expiration date or password.

**Comment by Piedone** [comment]
You could do this by doing something along the lines of the following:
- Content part to create a link or multiple links.
- Links have a long random string in the query string but otherwise use the standard public URL of the content item (or you can use a different URL too, even a different controller).
- Store the expiration along with the random string in the part.
- Have an `AuthenticationHandler` implementation that detects if the random string parameter is present in the request, checks if the currently authorized content item has that string stored, and succeeds the authorization if it has, fails it otherwise.
- Bonus: Register a "noindex, nofollow" meta tag with `IResourceManager` when authorization succeeds, so search engine bots don't index such pages.

Do you have concrete questions?
